adding disable_plugins options to yum module

mpeters opened the pull request:
This branch adds yum options for `noplugins` and `disableplugin`. `disableplugin` behaves very similar to `disablerepo` and accepts a comma separated list of plugins to disable.

Commit:
adding noplugins and disableplugins options to yum module

mpdehaan commented:
I'd prefer to see this as disable_plugins with a special value of "all".

We require variables to have "underscores_between_things" basically.

Various functions are also getting a little long on the argument signature and may wish to move to key-value arguments

If you'd like to update this first part, we'd appreciate it, and we can probably take care of the latter at a later date.

mpeters commented:
> I'd prefer to see this as disable_plugins with a special value of "all".

That makes sense.

> We require variables to have "underscores_between_things" basically.

I actually prefer this too, but I was trying to follow the original conventions of this module (and the yum command line tool itself) which uses `disablerepo`, `enablerepo`.

Commit:
Fixing issues mentioned in issue #7426 by mpdehaan
+ Renamed disableplugin to disable_plugin
+ Removed noplugins in favor of "all" value to disable_plugin

Commit:
adding an example of disable_plugin

mpeters commented:
Changes made.

Commit:
make plural as requested

amenonsen commented:
@mpeters any chance you could rebase this and submit it to the `ansible-modules-core` repositorry? This PR (against the `ansible` repository) should be closed.

gregdek commented:
Yep, this should be against ansible-core-modules. Closing.
